### DISCUSSION TITLE
Course 1 - Lesson 4 - at 10:05. On Differential Privacy in Netflix and IMDB example

### DISCUSSION THREAD

**Discussion by mmalekzadeh** [OpenMined/courses#195] [Course: Our Privacy Opportunity] [answered]
First, I don't know how to appreciate the time and effort that Andrew, Emma, and other amazing people in OpenMined have spent on preparing such a brilliant course.  I've decided to participate in discussions hoping that it would help others to better enjoy this course.

The specific question is that in [Course 1 - Lesson 4 - at time 10:05](https://youtu.be/XwtCeYdzTXc?t=603), Andrew discusses that:

> "Put in differential privacy terms, because Netflix released some epsilon and IMDB had  already released quite a big [epsilon] because it is a public website, so much epsilon was out in public that these researchers were able to successfully launch a de-anonymization attack."

My concern is that according to the differential privacy (DP) guarantee: *if somebody ensures that a data release is `epsilon-DP`, then technically nobody with any sort of auxiliary or side-channel information can break that guarantee*. This is an important aspect of DP that makes an `epsilon-DP` mechanism independent of any other information that would be available to an attacker.

Therefore, if Netflix could release some epsilon, then shouldn't we say that it does not matter how much information IMDB reveals, no one can use that IMDB information to violate Netflix's guarantee?

In my opinion, the reason that IMDB data helped to break Netflix's data anonymization is that Netflix didn't guarantee any sort of `epsilon-DP`. So, basically, Netflix didn't release any epsilon or maybe we can say that Netflix released an **infinite** epsilon.

Looking forward to hearing other people's opinions on this.
Best.

**Comment by hwrdtm** [comment, marked as answer]
This is interesting. I personally prefer to think about epsilon as a measure regarding whether an individual is identifiable when looking at a **single** flow of information (one data holder to one data analyst), rather than whether a data holder's privacy-guarantees are breached or not by another data holder's release. We can actually take your analogy further, to the point where IMDB releases an incredibly large amount of epsilon, and Netflix effectively releasing (close to) 0, where an individual can still be personally identifiable - in this case the claim that IMDB _breached_ Netflix's privacy guarantees is not applicable. Indeed, this side-channel attack is possible because IMDB's dataset is somewhat "similar" / "related" (loose use of terminology here), and for an analyst to make 1 analytic query from Netflix and 1 analytic query from IMDB, this is "similar" to that analyst making 2 queries from Netflix, thereby breaking some privacy budget, and thus enabling individuals to be identifiable.

I'm not clear regarding the specifics of Netflix's epsilon or whether they were working with some privacy budget, but I hope this framing can help understand the intricacies of epsilon.

**Reply by hershd23** [reply]
I sort with @hwrdtm on the definition of epsilon. I would like to add that these measures don't exactly quantify how much information about you is leaked, but rather how much more information can be leaked about you given what is already known about you.

The way that measures like epsilon and beta work are like if the probability of an event was say x and the dataset releases beta = 2 information about you the the new probability of that event cannot exceed 2x.

Now let's assume we added some noise to the Netflix dataset instead of anonymizing the dataset. Since IMDB data was already public the probability of finding the correct movie title given the review data was already high. On top of that when we query the differentially private data with say beta = 2, the guarantee that we get is that the new probability cannot be greater than 2 times the earlier probability.

This answer is based on my understanding of the course. Please feel free to correct me if required.
